### PR TITLE
Convert `imr-worldwide` to typescript

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
- convert third-party tag imr-worldwide to TypeScript

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Review by emma-imber — COMMENTED** [review]
Nice progress 📈 Maybe we could also delete the old flow file for this now?

File: static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts [hunk 1]
```diff
@@ -58,8 +58,10 @@ const guMetadata = {
 	'brand-only': 'P0EE0F4F4-8D7C-4082-A2A4-82C84728DC59',
 };
 
-const onLoad = () => {
-	const sectionFromMeta = config.get('page.section', '').toLowerCase();
+const onLoad = (): void => {
+	const sectionFromMeta = config
+		.get<string>('page.section', '')
```
Comment (Jakeii) [inline static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts:63]: would using the window.guardian.page object be better? Using the types specified there?

**Review by Jakeii — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts (hunk 1 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts:63]: Good idea!

File: static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { isInAuOrNz } from '../../../common/modules/commercial/geo-utils';
 
 // NOLCMB is a global function defined by the IMR worldwide library
 
-const guMetadata = {
+const guMetadata: Record<string, string> = {
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts:6]: I would recommend using the `as const` assertion, here!

Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/third-party-tags/imr-worldwide.ts:6]: I tried that originally but then it seemed like more effort than it was worth to then narrow `config.page.section`, a `string`, to `keyof typeof guMetadata` in order to use it to index into `guMetadata` (and I'm still not sure how to do this).

**Pull request merged**